src/S3/Result/ListObjectsV2Output.php

```diff
@@ -82,16 +84,31 @@ class ListObjectsV2Output extends Result implements \IteratorAggregate
      */
     public function getCommonPrefixes(bool $currentPageOnly = false): iterable
     {
-        $this->initialize();
-
         if ($currentPageOnly) {
+            $this->initialize();
+
             return $this->CommonPrefixes;
         }
+
+        if (!$this->client instanceof S3Client) {
+            throw new \InvalidArgumentException('missing client injected in paginated result');
+        }
+        $input = $this->lastRequest;
+        if (!$input instanceof ListObjectsV2Request) {
+            throw new \InvalidArgumentException('missing last request injected in paginated result');
+        }
+        $page = $this;
         while (true) {
-            yield from $this->CommonPrefixes;
+            $page->initialize();
+            yield from $page->CommonPrefixes;
```

Comment:
Im not sure if it is better or worse.. But one could do 
```php
yield from $page->getCommonPrefixes(true);
```

That would remove the need for `$page->initialize();`

Reply:
good catch, I'll fix that.